cachecontrol/caches/file_cache.py

```diff
@@ -122,15 +82,18 @@ def _write(self, path: str, data: bytes) -> None:
         Safely write the data to the given path.
         """
         # Make sure the directory exists
-        try:
-            os.makedirs(os.path.dirname(path), self.dirmode)
-        except OSError:
-            pass
+        dirname = os.path.dirname(path)
+        os.makedirs(dirname, self.dirmode, exist_ok=True)
 
         with self.lock_class(path + ".lock"):
             # Write our actual file
-            with _secure_open_write(path, self.filemode) as fh:
-                fh.write(data)
+            (fd, name) = tempfile.mkstemp(dir=dirname)
+            try:
+                os.write(fd, data)
```

Comment:
I don't think this is guaranteed to write all the data: https://docs.python.org/3/library/os.html#os.write

Reply:
Yep, I believe that's right. This is also true of the old code, since `fileobj.write(...)` can also perform a partial write.

I'll need to re-check, but I think the cases in which a partial write can occur here are pretty marginal when `fd` is a regular file: the main one that I remember is when the entire write would cause `ENOSPC`.